src/frontend/high_level_schedule.cpp

```diff
@@ -591,6 +591,24 @@ namespace graphit {
 
         }
 
+        high_level_schedule::ProgramScheduleNode::Ptr
+        high_level_schedule::ProgramScheduleNode::configParForGrainSize(std::string apply_label, int grain_size) {
+
+            if (schedule_ == nullptr){
+                schedule_ = new Schedule();
+            }
+
+            // If no par for schedule has been constructed, construct a new one
+            if (schedule_->par_for_grain_size_schedules == nullptr) {
+                schedule_->par_for_grain_size_schedules= new std::map<std::string, int>();
```

Comment:
Perhaps not so important for now because we have done it this way before, but we shouldn't use a pointer to a map. Just use a map. 
Maybe note this for future reference.